pylops/LinearOperator.py

```diff
@@ -253,12 +253,32 @@ def dot(self, x):
         elif np.isscalar(x):
             return _ScaledLinearOperator(self, x)
         else:
-            if x.ndim == 1:  # or x.ndim == 2 and x.shape[1] == 1:
-                return self.matvec(x)
+            dims = getattr(self, "dims", None)
```

Comment:
ultimately we should avoid this check by making sure we always have dims and dimsd (same as shape and dtype)

Reply:
I checked

```
In [7]: %timeit getattr(Mop, "dims", None)
60.5 ns ± 0.315 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [8]: %timeit Mop.dims
37.2 ns ± 1.3 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

though I am not sure if we care as it is an irrelevant time difference...